### DISCUSSION TITLE
Additional discount configuration: Shopify-like "once_per_customer" and "allocation_limit"

### DISCUSSION THREAD

**Discussion by eric-makro** [medusajs/medusa#3012] [Feature Requests]
Will Medusa Team have a plan to add Shopify-like "once_per_customer" and "allocation_limit" discount configuration.

- "once_per_customer"
It's used to set whether a discount will be valid only for a single use per customer

(please check below screenshot for Shopify reference)
<img width="950" alt="image" src="https://user-images.githubusercontent.com/110385557/212101329-fe9be9d3-d11a-4ab8-b564-6e80433805ab.png">

- "allocation_limit"
It's used to set the number of times the discount can be allocated on the cart if eligible

(please check below screenshot for Shopify reference)
<img width="995" alt="image" src="https://user-images.githubusercontent.com/110385557/212102052-89e3644d-0265-428e-973f-9efc8d2c6e90.png">

Thanks

**Comment by DidierGuyon** [comment]
@eric-makro For the `once_per_customer` rule, I created a pull request: https://github.com/medusajs/medusa/pull/3700
